lib/puppet/functions/find_linecard.rb

```diff
@@ -0,0 +1,32 @@
+#
+# Cisco find_linecard puppet manifest function.
+#
+# Copyright (c) 2018 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+Puppet::Functions.create_function(:find_linecard) do
```

Comment:
I see that you have changed these manifest functions to use the new puppet function infra.  Where these functions tested by applying the demo manifests?  We can verify the N7k specific functions but just curious about the functions that will work on N9k.

Reply:
Yea so during the testing of all the resources using the manifests from the examples/  it highlighted that the functions were in the old style, so once they got converted they have worked on our N9k VMs